store/tikv/region_cache.go

```diff
@@ -433,16 +449,32 @@ func (c *RPCContext) String() string {
 }
 
 type storeSelectorOp struct {
-	labels []*metapb.StoreLabel
+	labels          []*metapb.StoreLabel
+	excludedPeerIDs []uint64
 }
 
 // StoreSelectorOption configures storeSelectorOp.
 type StoreSelectorOption func(*storeSelectorOp)
 
-// WithMatchLabels indicates selecting stores with matched labels
+// WithMatchLabels indicates selecting stores with matched labels.
 func WithMatchLabels(labels []*metapb.StoreLabel) StoreSelectorOption {
 	return func(op *storeSelectorOp) {
-		op.labels = labels
+		if op.labels != nil {
+			op.labels = append(op.labels, labels...)
```

Comment:
You are able to do `append(nil slice type variable, ...)`, same for `excludedPeerIDs`.

Reply:
@JmPotato any reason you don't want to apply the patch? It is completely valid to do:

```
	var s []int = nil
	fmt.Println(append(s, 3))
```

Reply:
I have changed it, but not yet push.